pkg/virt-launcher/virtwrap/live-migration-source.go

```diff
@@ -511,14 +511,26 @@ func (l *LibvirtDomainManager) setMigrationResultHelper(vmi *v1.VirtualMachineIn
 	}
 
 	now := metav1.Now()
+	metaAbortStatus := domainSpec.Metadata.KubeVirt.Migration.AbortStatus
 	if abortStatus != "" {
-		metaAbortStatus := domainSpec.Metadata.KubeVirt.Migration.AbortStatus
 		if metaAbortStatus == string(abortStatus) && metaAbortStatus == string(v1.MigrationAbortInProgress) {
 			return domainerrors.MigrationAbortInProgressError
 		}
 		domainSpec.Metadata.KubeVirt.Migration.AbortStatus = string(abortStatus)
 	}
 
+	// In case that migrations are not reported as aborted, but as failed or succeeded, but we know that an abort was
+	// requested, we schould set the resulting outcome of the abort.
+	if isMigrationAbortInProgress(domainSpec) {
```

Comment:
I think we could move this below to each if statement. Also, let's check if we didn't set the `AbortStatus` already.
WDYT?

Reply:
The if statements below are not mutually exclusive. I could change the order below and move ti there, but I wanted to keep abortion separate from the rest of the metadata.

Regarding to checking on why we did not set it, that definitely makes sense!

Reply:
Not sure if this missing status is only related to this:

https://github.com/kubevirt/kubevirt/blob/1840e40b550d6b1e85ff96721b69233b7df5a964/pkg/virt-launcher/virtwrap/live-migration-source.go#L876

But maybe, in the `asyncMigrationAbort`, the abort status should be updated to MigrationAbortSucceeded there, it is only updating the status when failed.

Also, when libevirt has the status `DOMAIN_JOB_FAILED`, shouldn't we verify if it failed because the client request to abort (as done in line 812)? And if it was the case, we update the abort state?
https://github.com/kubevirt/kubevirt/blob/1840e40b550d6b1e85ff96721b69233b7df5a964/pkg/virt-launcher/virtwrap/live-migration-source.go#L842
https://github.com/kubevirt/kubevirt/blob/1840e40b550d6b1e85ff96721b69233b7df5a964/pkg/virt-launcher/virtwrap/live-migration-source.go#L812

Reply:
That can very well be. Great observations. I need a little bit to get back to this. @xpivarc meant he may be able to pick it up earlier than me. We will see :)

Reply:
>But maybe, in the asyncMigrationAbort, the abort status should be updated to MigrationAbortSucceeded there, it is only updating the status when failed.

Currently, the migration monitor is responsible to detect that abort was successful. We could have a discussion if it would be helpful to have this here but the monitor would still need to account for this situation.

>Also, when libevirt has the status DOMAIN_JOB_FAILED, shouldn't we verify if it failed because the client request to abort (as done in line 812)? And if it was the case, we update the abort state?

I think abort doesn't make the job failed.

There is a little bit of unknown as tracing the code with the given log suggests we should see the successful abort even without this patch. We need to have closer look.